explorer/src/utils/tx.ts

```diff
@@ -127,3 +133,33 @@ export function intoParsedTransaction(tx: Transaction): ParsedTransaction {
     },
   };
 }
+
+export function isSerumInstruction(instruction: TransactionInstruction) {
+  return instruction.programId.toBase58() === EXTERNAL_PROGRAMS["Serum"];
+}
+
+const SERUM_CODE_LOOKUP: { [key: number]: string } = {
+  0: "Initialize Market",
+  1: "New Order",
+  2: "Match Order",
+  3: "Consume Events",
+  4: "Cancel Order",
+  5: "Settle Funds",
+  6: "Cancel Order By Client Id",
+  7: "Disable Market",
+  8: "Sweep Fees",
+};
+
+export function parseSerumInstructionTitle(
+  instruction: TransactionInstruction
+): string {
+  try {
+    const code = instruction.data.slice(1, 5).readUInt32LE(0);
+
+    if (code in SERUM_CODE_LOOKUP) {
+      return SERUM_CODE_LOOKUP[code];
+    }
+  } catch (error) {}
```

Comment:
It'd be nice to know when this fails in Sentry, could you report this error?

Reply:
Yeah, was going to ask if we cared about that. Will add!